### PR TITLE
[14.0][MIG] purchase_order_approved

### PR DESCRIPTION
**Pull request by SurachetS**
standard migration

### CODE REVIEW

**Comment by kittiu** [comment]
No code changes. But when I test, the README says this,
Selection_098

But I can still change the value.
image

@SurachetS have you try that?

**Comment by SurachetS** [comment]
> No code changes. But when I test, the README says this,
> Selection_098
> 
> But I can still change the value.
> image
> 
> @SurachetS have you try that?

Fixed

**Review by PrapassornS — COMMENTED** [review]
Functional Tested LGTM.

**Comment by kittiu** [comment]
@HviorForgeFlow can you help verify if the above fix valid. As I test with v12 and v13 and it has the same issue, quite surprised.

image

Thank you.

File: purchase_order_approved/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    state = fields.Selection(selection_add=[("approved", "Approved")])
+    # TODO: inherit state but adding approved state in a position after 'to
+    # approve' state.
```
Comment (HviorForgeFlow) [inline purchase_order_approved/models/purchase_order.py]:
```suggestion
    state = fields.Selection(selection_add=[("approved", "Approved"), ("purchase",)])
```

**Comment by HviorForgeFlow** [comment]
It seems states inheritance is not working well for One2many field `order_line`.

@MiquelRForgeFlow any idea why? Omitting last push force it should work.

**Review by HviorForgeFlow — COMMENTED** [review]
Better this way

**Review by PrapassornS — APPROVED** [review]
Functional Tested LGTM.

File: purchase_order_approved/models/purchase_order.py (hunk 1 above)
Comment (SurachetS) [reply purchase_order_approved/models/purchase_order.py]: Fixed, Thank you

**Comment by SurachetS** [comment]
@kittiu @Saran440 please review

File: purchase_order_approved/models/purchase_order.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
-    state = fields.Selection(selection_add=[("approved", "Approved")])
+    state = fields.Selection(selection_add=[("approved", "Approved"), ("purchase",)])
     # TODO: inherit state but adding approved state in a position after 'to
     # approve' state.
```
Comment (HviorForgeFlow) [inline purchase_order_approved/models/purchase_order.py]: Drop those lines, TODO has been achieved

Comment (SurachetS) [reply purchase_order_approved/models/purchase_order.py]: fixed

File: purchase_order_approved/views/purchase_order_view.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                <button
+                    name="button_release"
+                    type="object"
+                    states="approved"
+                    string="Convert to Purchase Order"
+                    class="oe_highlight"
+                />
+            </button>
+            <field name="order_line" position="attributes">
+                <attribute
```
Comment (rousseldenis) [inline purchase_order_approved/views/purchase_order_view.xml:22]: @SurachetS @kittiu @HviorForgeFlow @JordiBForgeFlow Do you agree we need a purchase_stock_order_approved as the readonly should take into account service product as in:

https://github.com/odoo/odoo/blob/14.0/addons/purchase_stock/views/purchase_views.xml#L48

To be able to modify received quantities for service products

Comment (rousseldenis) [reply purchase_order_approved/views/purchase_order_view.xml:22]: Ok, found. The 'purchase' state should not be included in the readonly domain. I will do a PR for that

**Pull request merged**